polaris-react/src/components/TextField/TextField.tsx

```diff
@@ -90,6 +90,11 @@ interface NonMutuallyExclusiveProps {
   disabled?: boolean;
   /** Show a clear text button in the input */
   clearButton?: boolean;
+  /** Show an icon button within the input */
+  iconButton?: {
+    icon: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;
+    accessibilityLabel: string;
+  } | null;
```

Comment:
Let's use the `action` prop pattern to enable configuration in the same way as other component actions, like `Card.actions`

```suggestion
  /** An action rendered at the end of the input  */
 action?: ComplexAction | React.ReactNode;
```

Reply:
I think this is an example where we should be adding composition instead of new properties. Is there a way this can be built where we allow `React.ReactNode` and it's seen as a prefix or suffix?

Reply:
I agree with Alex here. That would make this more flexible. Would you imagine the usage be something like the following?

```tsx
<TextField suffix={iocnButtonMarkup} />
```

Reply:
Yep, exactly 💯 Checkout the sandbox demo mentioned [here](https://github.com/Shopify/polaris/pull/7810#pullrequestreview-1199748193)